src/dbus_fast/proxy_object.py

```diff
@@ -46,7 +57,7 @@ def __init__(self, bus_name, path, introspection, bus):
         self.path = path
         self.introspection = introspection
         self.bus = bus
-        self._signal_handlers = {}
+        self._signal_handlers: dict[str, list[SignalHandler]] = {}
```

Comment:
Need to use `typing` types for compatibility with older Python versions.

```suggestion
        self._signal_handlers: Dict[str, List[SignalHandler]] = {}
```

Reply:
this change seems unrelated to the rest of the PR though.

Reply:
> this change seems unrelated to the rest of the PR though.

Yes and no. I did it because this PR changed the type of `_signal_handlers`. It was previously type `dict[str, list[Callable]]` and after this PR it will be `dict[str, list[SignalHandler]]`. That was the reason for it, I wanted to make it clear what type it is now and that it had changed. But I am fine with dropping it if we want.

Also I did forget about the 3.7 rules, thanks for the reminder.